docs/production-readiness/develop.md

```diff
@@ -321,3 +322,236 @@ The information you are looking for may be found in the [legacy docs](https://le
+
+Note that with self-hosted Temporal Clusters, you must explicitly configure authorization specifications for the Web UI and CLI.
+
+The following samples provide implementation examples for applying authentication on your Codec Server using the Go SDK.
+
+- [Codec Server](https://github.com/temporalio/samples-go/tree/main/codec-server)
+- [GRPC proxy server](https://github.com/temporalio/samples-go/tree/main/grpc-proxy)
+
+#### Sample Request/Response
+
+Consider the following sample Request/Response when creating and hosting a Codec Server with the following specifications:
+
+- Scheme: https
+- Host: dev.mydomain.com/codec
+- Path: /decode
+
+```json
+HTTP/1.1 POST /decode
+Host: https://dev.mydomain.com/codec
+Content-Type: application/json
+X-Namespace: myapp-dev.acctid123
+
+{"payloads":[{"metadata":{"encoding":"anNvbi9wcm90b2J1Zg==","messageType":"dGVtcG9yYWxfc2hvcC5vcmNoZXN0cmF0aW9ucy52MS5TdGFydFNob3BwaW5nQ2FydFJlcXVlc3Q="},"data":"eyJjYXJ0SWQiOiJleGFtcGxlLWNhcnQiLCJzaG9wcGVySWQiOiJ5b3VyLXNob3BwZXItaWQtZXhhbXBsZSIsImVtYWlsIjoieW91ci1lbWFpbEBkb21haW4uY29tIn0"}]}
+
+200 OK
+Content-Type: application/json
+{
+  "payloads": [{
+    "metadata":{
+"encoding": "json/protobuf",
+"messageType": "temporal_shop.orchestrations.v1.StartShoppingCartRequest"
+    },"data":{
+"cartId":"example-cart",
+"shopperId":"your-shopper-id-example",
+"email":"your-email@domain.com"
+    }}]
+}
+```
+
+### Hosting your Codec Server
+
+Your Codec Server can be hosted at an organization-level or locally.
+
+#### Organization-level hosting
+
+Hosting the Codec Server for your organization simplifies both key management used for decryption and versioning the codec itself.
+Consider the following details for a multi-tenant approach to setting up your Codec Server:
+
+- Ingress: Your server must be reachable from the public internet, and will require configuration for ingress to that server.
+- Authorization: You must set explicit authorization checks to validate requests to your Codec Server.
+
+#### Local Hosting
+
+Locally hosting the Codec Server is simpler to get started.
+However, consider the following before choosing to do so:
+
+- A single URL configuration is accepted for the Cloud account.
+  This means some agreed-upon policy on the url must be made for everyone using the namespaces in this account.
+  For example, if you configure your Remote Codec Endpoint to be http://localhost:8080/codec then every developer must host your codec server locally at that port.
+
+  Alternatively, you can use the local hosts file to allow each developer to choose where to host.
+  For example, configure the remote codec endpoint as http://codec.server and allow each developer to control what it maps to locally.
+
+- Distributing encryption keys that can decrypt the payloads at your organization is a security risk.
+
+### Set your Codec Server endpoints with Web UI and CLI
```

Comment:
Can we please put a fork in this documentation road between Cloud and OSS ? This document is applicable for Cloud users right up until we don't show the screenshots for our Cloud web UI configuration and make certain that Cloud users know how to set it up (for example, Cloud users won't touch any kind of YAML config )